src/main/webapp/js/finscholar/commonlistview/commonlistview.soy

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{namespace finscholar.commonlistview.templates}
+
+{template .commonlistview}
+  {@param pagetype : string}
+  <table class="mdl-data-table mdl-js-data-table 
+      mdl-data-table--selectable mdl-shadow--2dp">
+    <thead>
+      {if $pagetype == 'scholarships'}
+        {call .scholarshiplistheader /}
+      {elseif $pagetype == 'colleges'}
+        {call .collegelistheader /}
+      {/if}
+    </thead>
+    <tbody id="table-body">
+    </tbody>
+  </table>
+  <div id="status">
+    {call .endoflist /}
+  </div>
+{/template}
+
+{template .scholarshiplistheader}
+  <tr id="title">
+    <th class="mdl-data-table__cell--non-numeric">Scholarship Name</th>
+    <th>Schools</th>
+    <th>Amount</th>
+  </tr>
+{/template}
+
+{template .collegelistheader}
+  <tr id="title">
+    <th class="mdl-data-table__cell--non-numeric">College Name</th>
+    <th>Acceptance Rate</th>
+    <th>Average ACT</th>
+  </tr>
+{/template}
+
+{template .listitems}
+  {@param batchofitems: [type: string,  
+                         items: list<list<string>>]}
+  {for $propsofoneitem in $batchofitems.items}
+    <tr class={$batchofitems.type} id={$propsofoneitem[0]}
+        jsaction="scholarshiplistview.clickAction;
+        dblclick:scholarshiplistview.doubleClickAction">
+    {for $index in range(1, $propsofoneitem.length())}
+      <td class="mdl-data-table__cell--non-numeric">{$propsofoneitem[$index]}</td>
+    {/for}
```

Comment:
Now each scholarship/college is a list. In the list the first field is id, the second field is name.
I'm not sure whether this method is clear enough for production.